scenarios/test_codegen/test/lib_tests/SourceManager_test.res

```diff
@@ -476,7 +522,11 @@ describe("SourceManager fetchNext", () => {
   })
 
   Async.it("Can add new partitions until the concurrency limit reached", async () => {
-    let sourceManager = SourceManager.make(~maxPartitionConcurrency=3, ~logger=Logging.logger)
+    let sourceManager = SourceManager.make(
+      ~sources=[source],
+      ~maxPartitionConcurrency=3,
+      ~logger=Logging.logger,
+    )
```

Comment:
Maybe worth adding a mock function and calling at each of these places rather than inliniing all of them